src/connectors/snowflake/trulens/connectors/snowflake/utils/snowflake_server_side_evalutation_artifacts.py

```diff
@@ -0,0 +1,202 @@
+        database_args=database_args,
+    )
+    session.start_evaluator(
+        run_location=FeedbackRunLocation.SNOWFLAKE,
+        return_when_done=True,
+        disable_tqdm=True,
+    )
+"""
+
+
+class SnowflakeServerSideEvaluationArtifacts:
+    def __init__(
+        self,
+        session: Session,
+        database_name: str,
+        schema_name: str,
+        warehouse_name: str,
+        external_access_integration_name: str,
+        database_url: str,
+    ):
+        self._session = session
+        self._database_name = database_name
+        self._schema_name = schema_name
+        self._warehouse_name = warehouse_name
+        self._external_access_integration_name = (
+            external_access_integration_name
+        )
+        self._database_url = database_url
+        self._validate_name(database_name, "database_name")
+        self._validate_name(schema_name, "schema_name")
+        self._validate_name(warehouse_name, "warehouse_name")
+        self._validate_name(
+            external_access_integration_name, "external_access_integration_name"
+        )
+
+    @staticmethod
+    def _validate_name(name: str, error_message_variable_name: str):
+        if not re.match(r"^[A-Za-z0-9_]+$", name):
+            raise ValueError(
+                f"`{error_message_variable_name}` must contain only alphanumeric and underscore characters!"
+            )
+
+    def set_up_all(self):
+        self._set_up_stage()
+        self._set_up_stream()
+        self._set_up_secret()
+        self._set_up_network_rule()
+        self._set_up_external_access_integration()
+        self._set_up_stored_procedure()
+        self._set_up_task()
+
+    def _set_up_stage(self):
+        self._session.sql(
+            "CREATE STAGE IF NOT EXISTS TRULENS_PACKAGES_STAGE"
+        ).collect()
+        data_directory = os.path.join(
+            os.path.dirname(__file__), "../../../data/snowflake_stage_zips"
+        )
+        for zip_to_upload in _ZIPS_TO_UPLOAD:
+            self._session.file.put(
+                os.path.join(data_directory, zip_to_upload),
+                "@TRULENS_PACKAGES_STAGE",
+            )
+
+    def _set_up_stream(self):
+        # TODO(dkurokawa): Make sure the stream doesn't keep reading old data
+        self._session.sql(
+            f"""
+            CREATE STREAM IF NOT EXISTS TRULENS_FEEDBACK_EVALS_STREAM
+                ON TABLE {self._database_name}.{self._schema_name}.TRULENS_FEEDBACKS
+            """
+        ).collect()
+
+    def _set_up_secret(self):
+        self._session.sql(
+            """
+            CREATE SECRET IF NOT EXISTS TRULENS_DB_URL
+                TYPE = GENERIC_STRING
+                SECRET_STRING = ?
+            """,
+            params=[self._database_url],
+        ).collect()
+
+    def _set_up_network_rule(self):
+        self._session.sql(
+            """
+            CREATE NETWORK RULE IF NOT EXISTS TRULENS_DUMMY_NETWORK_RULE
```

Comment:
What is the authorization required to create this network rule?

Reply:
It's just something in the schema, so if you can build the schema you can create the rule.